client/src/test/java/org/asynchttpclient/test/EchoHandler.java

```diff
@@ -69,6 +69,13 @@ public void handle(String pathInContext, Request request, HttpServletRequest htt
         httpResponse.sendRedirect(httpRequest.getHeader("X-redirect"));
         return;
       }
+      if (headerName.startsWith("X-fail")) {
+        httpResponse.setStatus(HttpServletResponse.SC_EXPECTATION_FAILED);
+        httpResponse.getOutputStream().write("custom error message".getBytes());
```

Comment:
This response is invalid: as you set a body (and we want one as we want to test this use case), you need to set the Content-Length.

Reply:
`String#getBytes` is something horrible that should be dropped from Java. Please pass the Charset, eg `getBytes(StandardCharsets.US_ASCII/UTF_8)`